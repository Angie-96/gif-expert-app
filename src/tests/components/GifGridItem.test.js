import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("GifGridItem Tests", () => {
  const title = "Example";
  const url = "https://localhost:3000/algo.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("should show GifGridItem correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have a title paragraph", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("img tag should have an url and alt", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("should have the animate__fadeIn class", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
