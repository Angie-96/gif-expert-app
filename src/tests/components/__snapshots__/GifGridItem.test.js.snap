// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Tests should show GifGridItem correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-1s"
>
  <img
    alt="Example"
    src="https://localhost:3000/algo.jpg"
  />
  <p>
    Example
  </p>
</div>
`;
